src/privatesend/privatesend-client.cpp

```diff
@@ -1434,83 +1434,75 @@ bool CPrivateSendClientSession::MakeCollateralAmounts(const CompactTallyItem& ta
-        // Change output will be large enough to be valid as a collateral or a source input for another run
-        nCollateralAmount = CPrivateSend::GetMaxCollateralAmount();
-    } else {
-        // Change output might be too small for another collateral if we try to create the largest collateral
-        // here, create a slightly smaller one instead
-        nCollateralAmount = CPrivateSend::GetMaxCollateralAmount() - CPrivateSend::GetCollateralAmount();
-    }
-    vecSend.push_back((CRecipient){scriptCollateral, nCollateralAmount, false});
-
-    // try to use non-denominated and not mn-like funds first, select them explicitly
-    CCoinControl coinControl;
-    coinControl.fAllowOtherInputs = false;
-    coinControl.fAllowWatchOnly = false;
-    coinControl.nCoinType = CoinType::ONLY_NONDENOMINATED;
-    // send change to the same address so that we were able create more denoms out of it later
-    coinControl.destChange = tallyItem.txdest;
-    for (const auto& outpoint : tallyItem.vecOutPoints) {
-        coinControl.Select(outpoint);
-    }
-
-    bool fSuccess = mixingWallet->CreateTransaction(vecSend, wtx, reservekeyChange,
-        nFeeRet, nChangePosRet, strFail, coinControl);
-    if (!fSuccess) {
-        LogPrint(BCLog::PRIVATESEND, "CPrivateSendClientSession::MakeCollateralAmounts -- ONLY_NONDENOMINATED: %s\n", strFail);
-        // If we failed then most likely there are not enough funds on this address.
-        if (fTryDenominated) {
-            // Try to also use denominated coins (we can't mix denominated without collaterals anyway).
-            coinControl.nCoinType = CoinType::ALL_COINS;
-            if (!mixingWallet->CreateTransaction(vecSend, wtx, reservekeyChange,
-                    nFeeRet, nChangePosRet, strFail, coinControl)) {
-                LogPrint(BCLog::PRIVATESEND, "CPrivateSendClientSession::MakeCollateralAmounts -- ALL_COINS Error: %s\n", strFail);
-                reservekeyCollateral.ReturnKey();
-                return false;
-            }
-        } else {
-            // Nothing else we can do.
-            reservekeyCollateral.ReturnKey();
-            return false;
-        }
+    } else if (txBuilder.TryAddOutputs({CPrivateSend::GetCollateralAmount(), CPrivateSend::GetCollateralAmount()})) {
+        nCase = 2;
+        // <case2>, see TransactionRecord::decomposeTransaction
+        // Out1 CPrivateSend::IsCollateralAmount()
+        // Out2 CPrivateSend::IsCollateralAmount()
+
+        // First add two outputs to get the available value after all fees
+        CTransactionBuilderOutput* out1 = txBuilder.AddOutput();
+        CTransactionBuilderOutput* out2 = txBuilder.AddOutput();
+
+        // Create two equal outputs from the available value. This adds one duff to the fee if txBuilder.GetAmountLeft() is odd.
+        CAmount nAmountOutputs = getAdjustedAmount(txBuilder.GetAmountLeft() / 2);
```

Comment:
Why would we want to create two equal outputs at GetCollateralAmount() + x instead of one output at GetCollateralAmount() and one at GetCollateralAmount + 2x? the latter seems preferable to me and more in line with current logic

Reply:
The idea behind the two equal outputs was just trying to reduce chance of false positives for this case in `TransactionRecord::decomposeTransaction` to best avoid something like #3597. 

Given the input amount is random and therefor `x` is random

- With 2x `GetCollateralAmount() + x` you are doing `amount1 == amount2 && IsCollateralAmount(amount1)` 

- With  `GetCollateralAmount()` and `GetCollateralAmount() + 2x` you are doing `amount1 == GetCollateralAmount() && IsCollateralAmount(amount2)`

For me the first bullet point, especially the `amount1 == amount2` with the random `x` included just looked to me like a better solution to prevent false positives in this case. And as i couldn't see any drawbacks i implemented it like this.

Thoughts?